src/app/CommandSender.cpp

```diff
@@ -130,7 +130,10 @@ CHIP_ERROR CommandSender::OnMessageReceived(Messaging::ExchangeContext * apExcha
 
     if (mState == State::AwaitingTimedStatus)
     {
-        err = HandleTimedStatus(aPayloadHeader, std::move(aPayload));
+        CHIP_ERROR statusError = CHIP_NO_ERROR;
+        SuccessOrExit(err = StatusResponse::ProcessStatusResponse(std::move(aPayload), statusError));
```

Comment:
This needs to first check that the message is in fact a StatusResponse, no?  This is why there was a helper method for this not-entirely-trivial logic in `TimedRequest`....

Reply:
yes, just add this check back. thanks for detailed check.